### PR TITLE
Added Unit and Cypress E2E tests for existing components

### PR DESCRIPTION
**Pull request by prateekd15**
## Purpose
Added unit test cases and e2e test cases for below three components as per conditions

### Counter
* Test that component renders initial value provided in props
* Test that a click event on "decrement" button decrements the displayed value
* Test that a click event on "increment" button increments the displayed value

### SearchForm
* Test that component renders an input with the value equal to initial value passed in props
* Test that after typing to the input and a "click" event on the Submit button, the "onChange" prop is called with proper value
* Test that after typing to the input and pressing Enter key, the "onChange" prop is called with proper value

### GenreSelect
* Test that component renders all genres passed in props
* Test that component highlights a selected genre passed in props
* Test that after a click event on a genre button component calls "onChange" callback and passes correct genre in arguments

## Visuals
Jest unit tests
image

Cypress E2E tests
image

### CODE REVIEW

File: src/component/Counter/Counter.test.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { render, fireEvent } from "@testing-library/react";
+import Counter from "./Counter";
+
+describe("Counter Component", () => {
+  it("Should render with initial count", () => {
+    const { getByText } = render(<Counter initialValue={5} />);
```
Comment (Ajay-Nallanagula) [inline src/component/Counter/Counter.test.js]: You can also think of encapsulating render logic, thus also reducing redundancy
`
setUp(props){ 
  return render(<Counter {...props}}>)
}
  //Inside tests 
const {getBytext} = setUp(props)....
`

File: src/component/Counter/Counter.test.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  });
+
+  it("Should increment the count when the increment button is clicked", () => {
+    const { getByText } = render(<Counter initialValue={0} />);
+    const incrementButton = getByText("+");
+
+    fireEvent.click(incrementButton);
+
+    const counterElement = getByText("Counter: 1");
+    expect(counterElement).toBeInTheDocument();
```
Comment (Ajay-Nallanagula) [inline src/component/Counter/Counter.test.js]: Your Code: `expect(counterElement).toBeInTheDocument();`
Suggestion: Simplified shortcut can be just check `expect(counterElement)`
Reason: If there is no match  jest will throw an exception 

Plz refer: https://testing-library.com/docs/react-testing-library/cheatsheet
image

File: src/component/GenreButton/GenreButton.test.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { render, fireEvent } from "@testing-library/react";
+import GenreButton from "./GenreButton";
+
+describe("GenreButton", () => {
+  it("renders the button with the correct class and label", () => {
+    const genre = "Action";
+    const buttonClass = "red";
+
+    const { getByText, getByRole } = render(
+      <GenreButton genre={genre} buttonClass={buttonClass} onSelect={() => {}} />
```
Comment (Ajay-Nallanagula) [inline src/component/GenreButton/GenreButton.test.js]: render can be moved to setUp() as suggested above.

**Review by Ajay-Nallanagula — CHANGES_REQUESTED** [review]
Left few minor comments

File: src/component/Counter/Counter.test.js (hunk 1 above)
Comment (prateekd15) [reply src/component/Counter/Counter.test.js]: Fixed as part of 903b467

File: src/component/Counter/Counter.test.js (hunk 2 above)
Comment (prateekd15) [reply src/component/Counter/Counter.test.js]: Fixed as part of 903b467

File: src/component/GenreButton/GenreButton.test.js (hunk 3 above)
Comment (prateekd15) [reply src/component/GenreButton/GenreButton.test.js]: Fixed as part of 903b467

**Review by Ajay-Nallanagula — APPROVED** [review]
Good Job!!

**Pull request merged**